suite-common/wallet-core/src/discovery/discoveryThunks.ts

```diff
@@ -615,6 +619,10 @@ export const startDiscoveryThunk = createThunk(
                 // try to generate device metadata master key
                 await dispatch(initMetadata(false));
             }
+
+            // NOTE: here we know what's the status of the wallet auth confirmed or not.
+            extra.services.discoveryHook.triggerDiscoveryAuthHooks(deviceState, device);
```

Comment:
This is ugly as hell , but it is just a hotfix. We should be able to run the discovery on-demand regardless of which device is selected.